### PR TITLE
Expiration support

### LINKED ISSUE #5: Support expiration to force regeneration of tests
The idea is that you would be able to set an option to say "expire the recordings after X days/duration" which would force re-generation of the recordings after that time has passed.

This is to ensure that if the 3rd party API you're using suddenly makes unexpected breaking changes, you could catch them.

### PR DESCRIPTION
**Pull request by exoego**
Closes #5

* `RecordOptions` now have `expiresAfter: Option[java.time.Duration]`: an option to say "expire the recordings after X days/duration" 
* `VcrRecords` now have `expiration: Option[OffsetDateTime]`
    * which is calculated "the datetime of saving file + `expiresAfter` duration"
    * `VcrClient` ignores a previous record file if it is expired.

### CODE REVIEW

File: core/src/main/scala/bekindrewind/RecordOptions.scala [hunk 1]
```diff
@@ -1,7 +1,12 @@
 package bekindrewind
 
-final case class RecordOptions(notRecordedThrowsErrors: Boolean, overwriteAll: Boolean) {
+import scala.concurrent.duration.FiniteDuration
 
+final case class RecordOptions(
+  notRecordedThrowsErrors: Boolean,
+  overwriteAll: Boolean,
+  expiresAfter: Option[FiniteDuration]
```
Comment (reibitto) [inline core/src/main/scala/bekindrewind/RecordOptions.scala:8]: This works, but I'd prefer to use `java.time` exclusively (`java.time.Duration` in this case).

**Review by reibitto — CHANGES_REQUESTED** [review]
Thanks! This looks good. Just 1 minor thing.

File: core/src/main/scala/bekindrewind/RecordOptions.scala (hunk 1 above)
Comment (exoego) [reply core/src/main/scala/bekindrewind/RecordOptions.scala]: Changed to `java.time.Duration` in https://github.com/reibitto/be-kind-rewind/pull/18/commits/6d3d8ee5fb2dea9571c1772dd47daeb31e3fa4a6

**Review by reibitto — APPROVED** [review]
Great! Thanks again! 😀

**Pull request merged**